Add support packages, cache the MATLAB installs, and add a few triggers

acampbel opened the pull request.

davehoadley reviewed (commented):
@acampbel thanks, that looks great! I did note the upload mltbx stage of the workflow had a warning about file not found and it seems to be missing from your branch. Or, would I only see it in the artifacts download?

acampbel commented:
> @acampbel thanks, that looks great! I did note the upload mltbx stage of the workflow had a warning about file not found and it seems to be missing from your branch. Or, would I only see it in the artifacts download?

nope! the warning was valid. It was because I didn't update the default build task to be the "toolbox" task, but I have fixed that. Note as part of this I actually deleted the mltbx from your source control because it is now a "derived" file.

Also, I am not sure that it contains really what you want in the toolbox...it probably contains too much of the project. there may be some slight restructuring of the project to help only include what you want in the mltbx deployable.

also, was the self hosted runner only in place because of the support package use case? If so then I think you can get rid of that. Do you want to run tests on all platforms? Running locally on my Mac fails, complaining about one fo the mat files not being valid mat file. Is that just a platform thing?

davehoadley commented:
> > @acampbel thanks, that looks great! I did note the upload mltbx stage of the workflow had a warning about file not found and it seems to be missing from your branch. Or, would I only see it in the artifacts download?
> 
> nope! the warning was valid. It was because I didn't update the default build task to be the "toolbox" task, but I have fixed that. Note as part of this I actually deleted the mltbx from your source control because it is now a "derived" file.
> 
> Also, I am not sure that it contains really what you want in the toolbox...it probably contains too much of the project. there may be some slight restructuring of the project to help only include what you want in the mltbx deployable.
> 
> also, was the self hosted runner only in place because of the support package use case? If so then I think you can get rid of that. Do you want to run tests on all platforms? Running locally on my Mac fails, complaining about one fo the mat files not being valid mat file. Is that just a platform thing?

* The self-hosted runner was 100% because of the addons installation issue. I can drop it. 
* I'll check the toolbox output and see if it contains what I really want.
* Cross-platform support is intended. Your .mat file issue is because I am using LFS for the large mat files. For simplicity I put them all in LFS with my .gitattributes. You'll need to do a git lfs checkout after you clone to have the tests pass.

acampbel commented:
> > > @acampbel thanks, that looks great! I did note the upload mltbx stage of the workflow had a warning about file not found and it seems to be missing from your branch. Or, would I only see it in the artifacts download?
> > 
> > 
> > nope! the warning was valid. It was because I didn't update the default build task to be the "toolbox" task, but I have fixed that. Note as part of this I actually deleted the mltbx from your source control because it is now a "derived" file.
> > Also, I am not sure that it contains really what you want in the toolbox...it probably contains too much of the project. there may be some slight restructuring of the project to help only include what you want in the mltbx deployable.
> > also, was the self hosted runner only in place because of the support package use case? If so then I think you can get rid of that. Do you want to run tests on all platforms? Running locally on my Mac fails, complaining about one fo the mat files not being valid mat file. Is that just a platform thing?
> 
> * The self-hosted runner was 100% because of the addons installation issue. I can drop it.
> * I'll check the toolbox output and see if it contains what I really want.
> * Cross-platform support is intended. Your .mat file issue is because I am using LFS for the large mat files. For simplicity I put them all in LFS with my .gitattributes. You'll need to do a git lfs checkout after you clone to have the tests pass. I'll a note about that to readme.md

I updated the ci to run on all four platforms (linux, windows, intel Mac and apple silicon) may be overkill but you can remove platforms you dont need. As an aside it was a bit of a nightmare dealing with LFS on my Mac. I got it to a point where it was half working, but not completely and for the time being I gave up on it. This may be user error, but it also seems like MATLAB on the Mac and the associated project support might have some challenges.
